controller/state/statedb.go

```diff
@@ -1078,3 +1083,29 @@ func (s *stateDB) ResetWorkerTasks(ctx context.Context, worker string) error {
 	}
 	return nil
 }
+
+func (s *stateDB) Delete(ctx context.Context, uuid string) error {
+	task, _, err := s.getWithTag(ctx, uuid)
```

Comment:
It might be better to create a version of this function that is not transaction-wrapped and call to get and delete within the same transaction.

Reply:
agree.

transaction wrapping should happen in the calling code for a utility function like this, IMHO.